### PR TITLE
kernel: precise trigger timing of post-fs-data

### PR DESCRIPTION
**Pull request by Ylarod**

### CODE REVIEW

**Comment by Ylarod** [comment]
```
[    0.419555] KernelSU: ksu_lsm_hook_init
[    0.422768] KernelSU: sucompat: execve_kp: 0
[    0.425883] KernelSU: sucompat: newfstatat_kp: 0
[    0.428921] KernelSU: sucompat: faccessat_kp: 0
[    0.431946] KernelSU: ksud: execve_kp: 0
[    0.434959] KernelSU: ksud: vfs_read_kp: 0
[    2.060064] KernelSU: /system/bin/init second_stage executed
[    2.160675] KernelSU: vfs_read: /system/etc/init/atrace.rc, comm: init, count: 1024, rc_count: 336
[    2.160809] KernelSU: unregister vfs_read kprobe: 1!
[    5.770245] KernelSU: option: 0xdeadbeef, cmd: 7
[    5.770254] KernelSU: post-fs-data triggered
[    5.770255] KernelSU: ksu_load_allow_list
[    5.770839] KernelSU: allowlist version: 1
[    5.770845] KernelSU: load_allow_uid: 2000, allow: 1
[    5.770847] KernelSU: load_allow_uid: 10268, allow: 1
[    5.770849] KernelSU: ksu_allow_uid :2000, allow: 1
[    5.770851] KernelSU: load_allow_uid: 10299, allow: 1
[    5.770852] KernelSU: ksu_allow_uid :2000, allow: 1
[    5.770854] KernelSU: ksu_allow_uid :10268, allow: 1
[    5.770856] KernelSU: load_allow_uid: 10234, allow: 1
[    5.770857] KernelSU: ksu_allow_uid :2000, allow: 1
[    5.770858] KernelSU: ksu_allow_uid :10268, allow: 1
[    5.770860] KernelSU: ksu_allow_uid :10299, allow: 1
[    5.770862] KernelSU: load_allow_uid: 10374, allow: 1
[    5.770863] KernelSU: ksu_allow_uid :2000, allow: 1
[    5.770864] KernelSU: ksu_allow_uid :10268, allow: 1
[    5.770866] KernelSU: ksu_allow_uid :10299, allow: 1
[    5.770867] KernelSU: ksu_allow_uid :10234, allow: 1
[    5.770869] KernelSU: load_allow_uid: 10419, allow: 0
[    5.770870] KernelSU: ksu_allow_uid :2000, allow: 1
[    5.770872] KernelSU: ksu_allow_uid :10268, allow: 1
[    5.770873] KernelSU: ksu_allow_uid :10299, allow: 1
[    5.770874] KernelSU: ksu_allow_uid :10234, allow: 1
[    5.770876] KernelSU: ksu_allow_uid :10374, allow: 1
[    5.770878] KernelSU: load_allow_list read err: 0
[    5.770901] KernelSU: save allow list uid :2000, allow: 1
[    5.770904] KernelSU: save allow list uid :10268, allow: 1
[    5.770907] KernelSU: save allow list uid :10299, allow: 1
[    5.770909] KernelSU: save allow list uid :10234, allow: 1
[    5.770912] KernelSU: save allow list uid :10374, allow: 1
[    5.770914] KernelSU: save allow list uid :10419, allow: 0
[    6.217544] KernelSU: exec app_process, /data prepared!
[    6.217552] KernelSU: on_post_fs_data already done
[    6.217557] KernelSU: unregister execve kprobe: 1!
[   13.262576] KernelSU: renameat: packages.list.tmp -> packages.list\x0a, new path: /system/packages.list
[   16.063023] KernelSU: newfstatat su->sh!
[   16.063065] KernelSU: faccessat su->sh!
[   16.063433] KernelSU: do_execveat_common su found
[   16.063449] KernelSU: error: 0, sid: 1171
[   20.028713] KernelSU: renameat: packages.list.tmp -> packages.list\x0a, new path: /system/packages.list
[   20.768997] KernelSU: option: 0xdeadbeef, cmd: 7
[   20.769005] KernelSU: boot_complete triggered
```

**Comment by tiann** [comment]
this may not work at first boot(ksud is not exist because manager is not installed)

**Comment by Ylarod** [comment]
The legacy way is also keeped. 

If we embed ksud to kernel, there comes another problem: we can write it to `/data/adb/ksud` only after `post-fs-data`, but we need ksud to be awared of `post-fs-data`. This seems to enter a infinite loop.

If init execve ksud without checking it's existance, we can sense this event in execve handler

**Comment by tiann** [comment]
if embed to kernel, just put it to /dev

**Comment by tiann** [comment]
which is earier between legacy way and new way？

**Comment by Ylarod** [comment]
new way
```
[    5.770254] KernelSU: post-fs-data triggered
[    6.217544] KernelSU: exec app_process, /data prepared!
[    6.217552] KernelSU: on_post_fs_data already done
```

**Comment by tiann** [comment]
please format the code

File: kernel/core_hook.c [hunk 1]
```diff
@@ -200,6 +202,33 @@ int ksu_handle_prctl(int option, unsigned long arg2, unsigned long arg3,
 		}
 	}
 
+	if (arg2 == CMD_REPORT_EVENT) {
+		if (0 == current_uid().val) {
```
Comment (tiann) [inline kernel/core_hook.c:206]:
```
if ( 0 != current_uid().val) {
        return 0;
}
```

**Review by tiann — CHANGES_REQUESTED** [review]

File: .github/workflows/build-kernel-5.10.yml [hunk 2]
```diff
@@ -102,6 +102,12 @@ jobs:
         echo "[+] Add KernelSU symbols"
         cat $KSU_ROOT/kernel/export_symbol.txt | awk '{sub("[ \t]+","");print "  "$0}' >> $SYMBOL_LIST
 
+    - name: Set boot sign key
```
Comment (tiann) [inline .github/workflows/build-kernel-5.10.yml:105]: What is this for? better move to a single pr

**Review by tiann — CHANGES_REQUESTED** [review]

File: .github/workflows/build-kernel-5.10.yml (hunk 2 above)
Comment (Ylarod) [reply .github/workflows/build-kernel-5.10.yml:105]: e, bad merge

**Review by tiann — APPROVED** [review]

**Pull request merged**